common-utils/src/common/deployment/__init__.py

```diff
@@ -356,125 +378,270 @@ class FlowSpec(BaseModel):
+            pyproject_metadata (PyProjectMetadata): metadata from pyproject.toml
+            account_id (str): AWS account id
+            ecs_client (object): valid ecs client
         """
-        project_name = self._project_name
-        ecs_block_name = self._create_ecs_task_block()
-        s3_block_name = f"{project_name}-{ENVIRONMENT_TYPE}-{DEPLOYMENT_TYPE}"
-        s3 = S3.load(s3_block_name)
-        LOGGER.info(f"Loaded S3 Block: {s3_block_name} for S3 Path: {s3.bucket_path}...")  # type: ignore
-        ecs = ECSTask.load(ecs_block_name)
-        LOGGER.info(f"Using ECS Block: {ecs.name}...")  # type: ignore
-        skip_upload = False
-        flow_folder = get_flow_folder(flow_path)
+
+        # init private attrs that are needed downstream
+        project_name = pyproject_metadata.project_name
+        slugified_fn_name = standard_slugify(self.flow.fn.__name__)
+        # setting flow name to "<subproject name>/<parent directory>/<flow function name>
+        flow_name = f"{project_name}.{get_flow_folder(flow_path)}.{slugified_fn_name}"
+
+        # update docker_env to proper image name
+        orig_docker_env = self.docker_env
+        docker_envs = pyproject_metadata.docker_envs
+        self.docker_env = create_image_name_str(
+            project_name,
+            orig_docker_env,
+            docker_envs[orig_docker_env]["python_version"],
+        )
+
+        # process deployments
+        # create ECS task definition and Prefect Block
+        slugified_flow_name = standard_slugify(self.flow.name)
+        ecs_block_name = self._create_ecs_task_block(
+            account_id, ecs_client, slugified_flow_name
+        )
+        LOGGER.info(f"Using ECS Block: {ecs_block_name}...")
+        # loop through deployments and push
         for d in self.deployments:
             d.push_deployment(
-                f"{s3_block_name}/{flow_folder}/{self._slugified_fn_name}",
-                ecs_block_name,  # type: ignore
+                project_name,
+                ecs_block_name,
                 flow_path,
                 self.flow.fn.__name__,
-                skip_upload,
+                flow_name,
             )
-            skip_upload = True
 
 
 class PrefectProject(BaseModel):
-    """Model used for fetching the project configuration and deploying docker envs and flows as needed."""
+    """Model used for fetching the project configuration to deploy docker envs and flows as needed."""
 
+    _pyproject_metadata: PyProjectMetadata = PrivateAttr()
     _project_docker_envs: list[FlowDockerEnv] = PrivateAttr()
     _prefect_flows_folder: DirectoryPath = PrivateAttr()
 
     def __init__(self, **data) -> None:
         """Set the private fields on intansiation."""
         super().__init__(**data)
+        # pull the pyproject.toml data needed for processing
+        self._pyproject_metadata = get_pyproject_metadata()
+        pyproject_metadata = self._pyproject_metadata
```

Comment:
here is where we grab the pyproject.toml metadata for the whole model and downstream calls